src/index.test.ts

```diff
@@ -38,6 +38,24 @@ test('Should perform an initial fetch', async () => {
     expect(isEnabled).toBe(true);
 });
 
+test('Should perform an initial fetch as POST', async () => {
+    fetchMock.mockResponseOnce(JSON.stringify(data));
+    const config: IConfig = {
+        url: 'http://localhost/test',
+        clientKey: '12',
+        appName: 'webAsPOST',
+        usePOSTrequests: true,
+    };
+    const client = new UnleashClient(config);
+    await client.start();
+    
+    const request = getTypeSafeRequest(fetchMock, 0);
+    const body = JSON.parse(request.body as string);
+
+    expect(request.method).toBe('POST');
+    expect(body.context.appName).toBe('webAsPOST');
+});
```

Comment:
Should we have the inverse test also? That not setting the property keeps the method as GET.

Reply:
yes makes sense.